### ISSUE TITLE
PEP 3127, PEP 237 Eliminate use of long literals

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8366]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8366|
|Type|enhancement|
|Created|2016-05-26 22:18:00Z|
|Branch|https://github.com/twisted/twisted/tree/8366-rodrigc-long|

In Python 2, this code is legal, but in Python 3, it gives a syntax error:

```
a = 235L
```

The correct syntax for integer literals is documented in PEP 3127:
https://www.python.org/dev/peps/pep-3127/

The unification of int and long types in Python 3 is documented here:
https://www.python.org/dev/peps/pep-0237/

<details><summary>Searchable metadata</summary>

```
trac-id__8366 8366
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8366_rodrigc_long 8366-rodrigc-long
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1464301080294875 1464301080294875
changetime__1465420812580377 1465420812580377
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/82

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban commented|
|-|-|

Changes look good.

I have updated from `compat.long(1)` to just `long(1)` as this is the form Amber was using in the other py3 ported code.

There was an error in inotify.py code but I have fixed it.

Waiting for test to be executed with my changes and will come back.

I am also waiting for the coverage report

Many thanks for the changes.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph set owner to @rodrigc|
|-|-|

Thanks for the fix, rodrigc!

I'm pretty sure that in _most_ of these cases you could avoid the use of `compat.long` entirely and simply specify an int literal; particularly in the cases of very long integer constants, simply deleting the `L` should be sufficient.  If possible, I'd prefer to avoid tighter coupling to `twisted.python.compat`, especially if it's not really necessary.  Many of these are probably left over from ancient versions of Python where the `L` suffix was necessary.

Can you try a version of this that only uses `compat.long` in the places where it's necessary to make the implementation pass the tests, and just use plain `int`s everywhere?  Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_6"></a>|@rodrigc set owner to @glyph|
|-|-|

OK, I uploaded a new diff which does not use compat.long and just removes the L suffix:

https://github.com/twisted/twisted/pull/142

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_7"></a>|@adiroiban set owner to @rodrigc|
|-|-|

Travis tests are green, so I hope that buildbot will also be green.

Waiting for buildbot results and will merge.

Many thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_8"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset dccc7c3fb0b6637f5e0a831a7e94cad2000d2383
```
#!CommitTicketReference repository="" revision="dccc7c3fb0b6637f5e0a831a7e94cad2000d2383"
Merge 8366-rodrigc-long: Eliminate the use of long literals.

Author: rodrigc
Reviewer: adiroiban
Fixes: #8366
```
